java/core/src/main/java/com/microsoft/bond/TaggedProtocolStreamBonded.java

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond;
+
+import com.microsoft.bond.helpers.ArgumentHelper;
+import com.microsoft.bond.protocol.ProtocolWriter;
+import com.microsoft.bond.protocol.TaggedProtocolReader;
+
+import java.io.IOException;
+
+/**
+ * An implementation of {@link Bonded} backed by a tagged protocol reader, which effectively
+ * implements delayed deserialization for tagged protocols.
+ *
+ * @param <T> the struct type of this Bonded instance or {@link BondSerializable} if unspecified
```

Comment:
So to construct an unspecified one, you would write `new TaggedProtocolStreamBonded<BondSerializable>(reader, null)`?